### DISCUSSION TITLE
langium custom Validator

### DISCUSSION THREAD

**Discussion by zrwsmd** [eclipse-langium/langium#1430] [Q&A]
i validate st program name should start with /^\d/.test(program.name) for eaxmple,.
if i use true instead of isStartsWithNumber variable,it  hint below message,but replace isStartsWithNumber  variable with true  did hint 
this message
![image](https://github.com/eclipse-langium/langium/assets/26995040/45754f5a-ddc6-49b0-9e2d-f43fe488d762)
![image](https://github.com/eclipse-langium/langium/assets/26995040/2fb999e1-03ae-49c2-af39-943af033e994)

**Comment by msujew** [comment]
I'm sorry, what? I don't really understand what the question/issue is. Can you elaborate a bit?

**Reply by zrwsmd** [reply]
if i write 
if (true) {
            accept('error', 'program name should not start with a digital', { node: program, property: 'name' });
        }
this custom validationmessage will come out like below screenshot(also have langium lexer and parse error )
![image](https://github.com/eclipse-langium/langium/assets/26995040/a9ec81ea-81f1-4135-acbb-45971d8cd707)


  if i write 
if (isStartsWithNumber) {
            accept('error', 'program name should not start with a digital', { node: program, property: 'name' });
        }
in only have langium lexer and parse error
![image](https://github.com/eclipse-langium/langium/assets/26995040/22f03445-29ee-45ca-aff7-c810e988411c)

**Reply by msujew** [reply]
Can you show the terminal that you use to parse the `name` property of the `Program` type? There might be a parser error that prevents the assignment of the `name` property.

**Reply by zrwsmd** [reply]
Namespace:
	'NAMESPACE' name=ID
		(elements+=AbstractElement)*
	'END_NAMESPACE'
;

terminal ID: /[a-zA-Z][\w_&]*/;

**Reply by msujew** [reply]
Your `name` property is never assigned, since the `ID` cannot start with a digit. Only a-z and A-Z are valid characters. That's why the error doesn't appear when you try to test for the first digit. `program.name` is simply undefined due to the parser error.

**Reply by zrwsmd** [reply]
so langium parse error and custom validation error can not appear at the same time?or how to fix it

**Reply by msujew** [reply]
Not if they are in the exact same token, no. If the parser expects an ID, and receives a number, it won't attempt to assign the number to the field that is supposed to be an ID.

The presence of parser errors will lead to non-valid ASTs. 

You fix it by allowing your IDs to start with numbers. Note that this will lead to a lexer conflict if you have a number terminal rule.
